superset/connectors/druid/models.py

```diff
@@ -563,11 +563,14 @@ def latest_metadata(self):
         """Returns segment metadata from the latest segment"""
         logging.info('Syncing datasource [{}]'.format(self.datasource_name))
         client = self.cluster.get_pydruid_client()
-        results = client.time_boundary(datasource=self.datasource_name)
-        if not results:
-            return
-        max_time = results[0]['result']['maxTime']
-        max_time = dparse(max_time)
+        try:
+            results = client.time_boundary(datasource=self.datasource_name)
+            if not results:
```

Comment:
Just noticed this now but we could fall back to `datetime.now()` when the function returns `None` as well

Reply:
@Mogball I updated the logic per your comment. PTAL.